plugins/module_utils/hetzner/api.py

```diff
@@ -0,0 +1,202 @@
+        return None
+
+    def get_zone_by_id(self, id):
+        """
+        Given a zone ID, return the zone contents if found.
+
+        @param id: The zone ID
+        @return The zone information (DNSZone), or None if not found
+        """
+        result, info = self._get('v1/zones/{id}'.format(id=id), expected=[200, 404], must_have_content=[200])
+        if info['status'] == 404:
+            return None
+        return _create_zone_from_json(result['zone'])
+
+    def get_zone_records(self, zone_id, prefix=NOT_PROVIDED, record_type=NOT_PROVIDED):
+        """
+        Given a zone ID, return a list of records, optionally filtered by the provided criteria.
+
+        @param zone_id: The zone ID
+        @param prefix: The prefix to filter for, if provided. Since None is a valid value,
+                       the special constant NOT_PROVIDED indicates that we are not filtering.
+        @param record_type: The record type to filter for, if provided
+        @return A list of DNSrecord objects, or None if zone was not found
+        """
+        result = self._list_pagination('v1/records', data_key='records', query=dict(zone_id=zone_id))
+        return filter_records(
+            [_create_record_from_json(record) for record in result],
+            prefix=prefix,
+            record_type=record_type,
+        )
+
+    def add_record(self, zone_id, record):
+        """
+        Adds a new record to an existing zone.
+
+        @param zone_id: The zone ID
+        @param record: The DNS record (DNSRecord)
+        @return The created DNS record (DNSRecord)
+        """
+        data = _record_to_json(record, include_type=True, zone_id=zone_id)
+        result, dummy = self._post('v1/records', data=data, expected=[200])
+        return _create_record_from_json(result['record'])
+
+    def update_record(self, zone_id, record):
+        """
+        Update a record.
+
+        @param zone_id: The zone ID
+        @param record: The DNS record (DNSRecord)
+        @return The DNS record (DNSRecord)
+        """
+        if record.id is None:
+            raise DNSAPIError('Need record ID to update record!')
+        data = _record_to_json(record, include_type=True, zone_id=zone_id)
+        result, dummy = self._put('v1/records/{id}'.format(id=record.id), data=data, expected=[200])
+        return _create_record_from_json(result['record'])
+
+    def delete_record(self, zone_id, record):
+        """
+        Delete a record.
+
+        @param zone_id: The zone ID
+        @param record: The DNS record (DNSRecord)
+        @return True in case of success (boolean)
+        """
+        if record.id is None:
+            raise DNSAPIError('Need record ID to delete record!')
+        dummy, info = self._delete('v1/records/{id}'.format(id=record.id), must_have_content=False, expected=[200, 404])
+        return info['status'] == 200
+
+
+def create_hetzner_argument_spec():
+    return ArgumentSpec(
+        argument_spec=dict(
```

Comment:
@felixfontein can we add alias values for `prefix` (_name_)  and `zone` (_zone_name`)?  
That would make `community.dns` become a drop-in replacement.

Reply:
There are some more differences:
1. `state` has no default;
2. the interface is currently quite similar to `route53`, so `overwrite: true` has be added in quite a few cases.

I'm wondering whether the interface of the `_record` module shouldn't be overhauled anyway. Maybe let's discuss this in a separate issue; including which aliases to offer (in deprecated form for migration, and potentially permanently?).